src/graphql.tsx

```diff
@@ -573,8 +573,16 @@ export default function graphql(
             // while loading, we should use any previous data we have
             assign(data, this.previousData, currentResult.data);
           } else {
-            assign(data, currentResult.data);
-            this.previousData = currentResult.data;
+            let dataToMerge = currentResult.data;
+            // if we have an error, include any previously cached data
+            if (error) {
+              let previousResult = this.queryObservable.getLastResult();
+              if (previousResult && !previousResult.error) {
+                dataToMerge = assign({}, previousResult.data, currentResult.data);
```

Comment:
I think it's pretty hard to predict what data you'd get here if there ever is any data in `currentResult`. I think it might be better to just return the previous result.